### PR TITLE
3.next - Add deprecations for core package.

### PR DESCRIPTION
**Pull request by markstory**
Add runtime warnings to deprecated methods in Core. `InstanceConfigTrait` and `StaticConfigTrait` had a ton of use so this is a fairly big pull request :cry:

Refs #11385

### CODE REVIEW

File: src/Core/Configure/Engine/PhpConfig.php [hunk 1]
```diff
@@ -93,6 +93,7 @@ public function read($key)
         if (!isset($config)) {
             throw new Exception(sprintf('Config file "%s" did not return an array', $key . '.php'));
         }
+        deprecationWarning('PHP configuration files should not set `$config. Instead return an array.');
```
Comment (saeideng) [inline src/Core/Configure/Engine/PhpConfig.php]:
```php
if(!isset($config)){
   throw new Exception....
}else{
    throw new Warning......
}
```
what this say !?

Comment (markstory) [reply src/Core/Configure/Engine/PhpConfig.php]: The code path for a file returning an array is on line 89 - just outside if this diff.

File: tests/TestCase/Cache/Engine/XcacheEngineTest.php [hunk 2]
```diff
@@ -75,7 +75,7 @@ public function tearDown()
      */
     public function testConfig()
     {
-        $config = Cache::engine('xcache')->config();
+        $config = Cache::getConfig('xcache')->getConfig();
```
Comment (saeideng) [inline tests/TestCase/Cache/Engine/XcacheEngineTest.php]: `Cache::engine()` did not deprecate 
I think your want to use 
`Cache::engine('xcache')->getConfig();`

Comment (markstory) [reply tests/TestCase/Cache/Engine/XcacheEngineTest.php]: Derp. You're right this is a mistake.

**Pull request merged**